ambry-tools/src/test/java/com.github.ambry/store/BlobIdTransformerTest.java

```diff
@@ -0,0 +1,388 @@
+    StoreCopier.Message output = transformer.transform(inputAndExpected.getInput());
+    verifyOutput(output, inputAndExpected.getExpected());
+
+    mockStoreKeyConverter.setThrowException(true);
+    inputAndExpected = new InputAndExpected(blobIdStringPairList.get(1), VALID_MESSAGE_FORMAT_INPUT_STREAM_IMPLS[0]);
+    try {
+      transformer.transform(inputAndExpected.getInput());
+      fail("Did not throw MockStoreKeyConverterException");
+    } catch (MockStoreKeyConverter.MockStoreKeyConverterException e) {
+      //exception expected
+    }
+
+    mockStoreKeyConverter.setThrowException(false);
+    inputAndExpected = new InputAndExpected(blobIdStringPairList.get(2), VALID_MESSAGE_FORMAT_INPUT_STREAM_IMPLS[0]);
+    output = transformer.transform(inputAndExpected.getInput());
+    verifyOutput(output, inputAndExpected.getExpected());
+  }
+
+  /**
+   * Tests creating the transformer with a null StoreKeyConverter
+   */
+  @Test
+  public void testNullStoreKeyConverter() throws IOException {
+    try {
+      new BlobIdTransformer(null, blobIdFactory, null);
+      fail("Did not throw NullPointerException");
+    } catch (NullPointerException e) {
+      //expected
+    }
+  }
+
+  /**
+   * Tests creating the transformer with a null StoreKeyFactory
+   */
+  @Test
+  public void testNullStoreKeyFactory() throws IOException {
+    try {
+      new BlobIdTransformer(mockStoreKeyConverter, null, null);
+      fail("Did not throw NullPointerException");
+    } catch (NullPointerException e) {
+      //expected
+    }
+  }
+
+  /**
+   * Tests using the transformer with null input to the transform method
+   * @throws Exception
+   */
+  @Test
+  public void testNullTransformInput() throws Exception {
+    try {
+      transformer.transform(null);
+      fail("Did not throw NullPointerException");
+    } catch (NullPointerException e) {
+      //expected
+    }
+  }
+
+  /**
+   * Tests using the transformer with StoreCopier.Message inputs that have null components
+   * @throws Exception
+   */
+  @Test
+  public void testNullComponentsTransformInput() throws Exception {
+    MessageInfo messageInfo =
+        new MessageInfo(createBlobId(VERSION_1_UNCONVERTED), 123, (short) 123, (short) 123, 0L);
+    InputStream inputStream = null;
+    //null inputStream
+    StoreCopier.Message message = new StoreCopier.Message(messageInfo, inputStream);
+    try {
+      transformer.transform(message);
+      fail("Did not throw NullPointerException");
+    } catch (NullPointerException e) {
+      //expected
+    }
+    //null messageInfo
+    message = new StoreCopier.Message(null, new ByteArrayInputStream(new byte[30]));
+    try {
+      transformer.transform(message);
+      fail("Did not throw NullPointerException");
+    } catch (NullPointerException e) {
+      //expected
+    }
+  }
+
+  private BlobId createBlobId(String hexBlobId) throws IOException {
```

Comment:
minor: don't think a function is required

Reply:
Not a fan of having `new BlobId(stringId, clusterMap)` everywhere instead.